app/merginprojectmodel.cpp

```diff
@@ -49,12 +62,28 @@ int MerginProjectModel::rowCount(const QModelIndex &parent) const {
     return mMerginProjects.count();
 }
 
-void MerginProjectModel::resetProjects()
+void MerginProjectModel::resetProjects(const ProjectList &merginProjects)
 {
     mMerginProjects.clear();
     beginResetModel();
-    mMerginProjects = mApi->projects();
+    mMerginProjects = merginProjects;
     endResetModel();
 
     emit merginProjectsChanged();
 }
+
+void MerginProjectModel::downloadProjectFinished(QString projectFolder, QString projectName)
+{
+    Q_UNUSED(projectFolder);
+    for (std::shared_ptr<MerginProject> project: mMerginProjects) {
+        if (project->name == projectName) {
+            beginResetModel();
+            project->pending = false;
+            project->status = ProjectStatus::UpToDate;
+            endResetModel();
```

Comment:
no need to use [begin/end]ResetModel() here: we are not resetting the whole model, just modifying one item's content. For that it's best to use `emit dataChanged(index, index)` with the index of the changed project. This will trigger more optimized view update (it will not need to delete and recreate all items)